### PR TITLE
Run `mleap` tests in cross version testing

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## What changes are proposed in this pull request?

- Exclude `mleap` from `dev/extra-ml-requirements.txt` that pins scikit-learn to < 0.23.0 and causes several tests to fail. As a workaround, install mleap in `dev/run-python-sagemaker-tests.sh` before running tests for mleap.
- Closes #4721

## How is this patch tested?

Existing tests

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: dev/run-python-sagemaker-tests.sh [hunk 1]
```diff
@@ -22,8 +22,8 @@ pytest tests/sagemaker/mock --large
 # TODO: split out sagemaker tests and move other spark tests to run-python-flavor-tests.sh
 pytest tests/spark --large  --ignore tests/spark/test_mleap_model_export.py
 
-# MLeap doesn't support spark 3.x (https://github.com/combust/mleap#mleapspark-version)
-pip install pyspark==2.4.5
+# MLeap doesn't support spark > 3.0.2: https://github.com/combust/mleap#mleapspark-version
+pip install pyspark==3.0.2 mleap
```
Comment (dbczumar) [inline dev/run-python-sagemaker-tests.sh:26]: A historic day! MLeap now supports PySpark? Does that mean we can revive https://github.com/mlflow/mlflow/pull/4721?

Comment (harupy) [reply dev/run-python-sagemaker-tests.sh:26]: Looks like it does:

https://github.com/combust/mleap#mleapspark-version

image

Yep, I think we can!

Comment (dbczumar) [reply dev/run-python-sagemaker-tests.sh:26]: Cool! Can we do that as part of this PR?

Comment (harupy) [reply dev/run-python-sagemaker-tests.sh]: @dbczumar Updated the RP!

File: mlflow/ml-package-versions.yml [hunk 2]
```diff
@@ -349,3 +349,24 @@ spark:
...
         find tests/spark_autologging/datasource -name 'test*.py' | xargs -L 1 pytest --large
       fi
+
+mleap:
+  package_info:
+    pip_release: "mleap"
+
+  models:
+    minimum: "0.15.0"
+    maximum: "0.17.0"
```
Comment (dbczumar) [inline mlflow/ml-package-versions.yml:359]: This should be `0.18.0` right?

Comment (harupy) [reply mlflow/ml-package-versions.yml]: yep, thanks!

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

File: pylintrc [hunk 3]
```diff
@@ -150,6 +150,8 @@ disable=print-statement,
         # Ignore errors raised against `ExceptionSafeClass` and `ExceptionSafeAbstractClass`
         invalid-metaclass,
         broad-except,
+        no-name-in-module
+        import-error
```
Comment (harupy) [inline pylintrc:154]: Added these to ignore the following lint errors:

```
mlflow/mleap.py (215,4): [E0611 no-name-in-module] No name 'version' in module 'mleap'
mlflow/mleap.py (215,4): [E0401 import-error] Unable to import 'mleap.version'
mlflow/mleap.py (216,4): [E0611 no-name-in-module] No name 'pyspark' in module 'mleap'
mlflow/mleap.py (216,4): [E0401 import-error] Unable to import 'mleap.pyspark.spark_support'
```

If `mleap.version` is not importable, an error should be thrown when `mlflow/mleap.py` is run.

**Pull request merged**